launcher/container_runner.go

```diff
@@ -621,17 +631,39 @@ func (r *ContainerRunner) Run(ctx context.Context) error {
 	return nil
 }
 
+func pullImageWithRetries(f func() (containerd.Image, error), retry func() backoff.BackOff) (containerd.Image, error) {
```

Comment:
This retry method seems very generic, with only the return in the func being passed being specific to pulling images.
Is it possible to make this more generic? Or less generic?

Reply:
The intention is to make specific to pulling images. The reason I abstract this method is to make unit tests easier so that I can mock containerd API. Since cotainerd will make a network call to an actual docker registry, I want to avoid it in unit tests.